dask/dataframe/methods.py

```diff
@@ -245,9 +245,16 @@ def cummax_aggregate(x, y):
         return x if x > y else y
 
 
+def set_column(df, name, value):
+    df = df.copy(deep=False)
+    df[name] = value
+    return df
+
+
 def assign(df, *pairs):
-    kwargs = dict(partition(2, pairs))
-    return df.assign(**kwargs)
+    for name, val in dict(partition(2, pairs)).items():
+        df = set_column(df, name, val)
+    return df
```

Comment:
We probably don't need the set_column function if it's only used here.  I recommend that we just implement this directly within assign

```python
def assign(...):
    df = df.copy(deep=False)
    for ...
        ...
    return df
```

I personally find that reducing indirection like this is easier on future maintenance

Reply:
Agreed - I implemented it without indirection the first time around, but noticed some buggy behavior in the merge benchmark.  The actual cause was unrelated to this - Changing back.